libraries/azure_policy_insights_query_result.rb

```diff
@@ -0,0 +1,79 @@
+require 'azure_backend'
+
+class AzurePolicyInsightsQueryResult < AzureResourceBase
+  name 'azure_policy_insights_query_result'
+  desc 'Lists a collection of Azure Policy Insights Query Results'
+  example <<-EXAMPLE
+    describe azure_policy_insights_query_result(policy_definition: 'de875639-505c-4c00-b2ab-bb290dab9a54', resource_id: '/subscriptions/80b824de-ec53-4116-9868-3deeab10b0cd/resourcegroups/jfm-winimgbuilderrg2/providers/microsoft.virtualmachineimages/imagetemplates/win1021h1') do
+      it { should exist }
+    end
+  EXAMPLE
+
+  def initialize(opts = {})
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+    super(opts)
+    @resource_provider = specific_resource_constraint('Microsoft.Authorization/policyDefinitions', @opts)
+    validate_parameters(required: %i(policy_definition resource_id))
+    @azure_resource_id = @opts.delete(:resource_id)
+    @policy_definition = @opts.delete(:policy_definition)
```

Comment:
Is this because of line 16 ?

Reply:
no, we need it for line 57, where we check the resource URI and it  internally forces us to not use any other parameter even though it is valid for resource